### PR TITLE
fix(MqttTrigger Node): Fix mqtt trigger duplicated triggers

### PR DESCRIPTION
**Pull request by tinti**
The on message callback should not be called inside on subscribe callback as the documentation example shows.

The side effect is many workflows being trigger for a single event.

https://www.npmjs.com/package/mqtt

### CODE REVIEW

**Comment by Joffcom** [comment]
Hey @tinti,

This PR looks interesting and I know we have had some issues reported about duplicated trigger events from the MQTT node so it sounds like this may fix those.

Did you find a way to reproduce duplicated trigger events to test if it works?

**Comment by tinti** [comment]
> Hey @tinti,

Hi @Joffcom 

> This PR looks interesting and I know we have had some issues reported about duplicated trigger events from the MQTT node so it sounds like this may fix those.
> 
> Did you find a way to reproduce duplicated trigger events to test if it works?

I can guarantee that it has solved the issue with my use case.

Reproducing at my environment is simple. By starting https://hub.docker.com/_/eclipse-mosquitto and add 3 clients sending events every 30 seconds each is enough to trigger duplicated events.

Without the patch multiple events are created. With it, there is no duplicated events.

File: packages/nodes-base/nodes/MQTT/MqttTrigger.node.ts [hunk 1]
```diff
@@ -137,30 +137,32 @@ export class MqttTrigger implements INodeType {
...
-							resolve(true);
-						});
 					});
 				});
 
+				client.on('message', (topic: string, message: Buffer | string) => {
+					// tslint:disable-line:no-any
+					let result: IDataObject = {};
+
+					message = message.toString() as string;
```
Comment (tinti) [inline packages/nodes-base/nodes/MQTT/MqttTrigger.node.ts:147]:
```suggestion
					message = message.toString();
```

**Comment by Joffcom** [comment]
Hey @tinti,

We have recently redone the MQTT trigger node so I suspect this isn't an issue anymore. I am going to mark this PR as closed but thanks for the time you spent on this.

**Comment by tinti** [comment]
Hi @Joffcom,

That is great news! Thanks for the reply!

**Pull request closed without merging**